zplugin-ros1/Cargo.toml

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (c) 2022 ZettaScale Technology
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+# which is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+#
+# Contributors:
+#   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+#
+[package]
+name = "zplugin-ros1"
+version = "0.0.1"
+authors = [
+	"Dmitrii Bannov <dmitrii.bannov@zettascale.tech>",
+]
+edition = "2021"
+repository = "https://github.com/ZettaScaleLabs/zenoh-plugin-ros1"
+homepage = "http://zenoh.io"
+license = " EPL-2.0 OR Apache-2.0"
+categories = ["network-programming"]
+description = "Zenoh plugin for ROS1"
+
+[lib]
+name = "zplugin_ros1"
+crate-type = ["cdylib", "rlib"]
+
+[features]
+no_mangle = ["zenoh-plugin-trait/no_mangle"]
+default = ["no_mangle"]
+
+[dependencies]
+cyclors = "0.1.5"
+derivative = "2.2.0"
+env_logger = "0.9.1"
+flume = "0.10.14"
+futures = "0.3.24"
+git-version = "0.3.5"
+hex = "0.4.3"
+lazy_static = "1.4.0"
+log = "0.4.17"
+regex = "1.6.0"
+serde = "1.0.147"
+serde_json = "1.0.85"
+async-global-executor = "2.3.1"
+rand = "0.8.5"
+
+# zenoh = { version = "0.6.0-beta.1", features = ["unstable"] }
+# zenoh = { git = "https://github.com/eclipse-zenoh/zenoh.git", features = ["unstable"] }
+zenoh = { path = "../../../zenoh/zenoh", features = ["unstable"] }
+
+# zenoh-ext = { version = "0.6.0-beta.1", features = ["unstable"] }
+# zenoh-ext = { git = "https://github.com/eclipse-zenoh/zenoh.git", features = ["unstable"] }
+zenoh-ext = { path = "../../../zenoh/zenoh-ext", features = ["unstable"] }
+
+# zenoh-core = { version = "0.6.0-beta.1" }
+# zenoh-core = { git = "https://github.com/eclipse-zenoh/zenoh.git" }
+zenoh-core = { path = "../../../zenoh/commons/zenoh-core" }
+
+# zenoh-plugin-trait = { version = "0.6.0-beta.1", default-features = false }
+# zenoh-plugin-trait = { git = "https://github.com/eclipse-zenoh/zenoh.git", default-features = false }
+zenoh-plugin-trait = { path = "../../../zenoh/plugins/zenoh-plugin-trait", default-features = false }
+
+# rosrust = "0.9"
+# rosrust = { git = "https://github.com/ZettaScaleLabs/rosrust.git", branch = "feature/fix_bugs" }
```

Comment:
`cargo report future-incompatibilities --id 1` returns:

```
The following warnings were discovered during the build. These warnings are an
indication that the packages contain code that will become an error in a
future release of Rust. These warnings typically cover changes to close
soundness problems, unintended or undocumented behavior, or critical problems
that cannot be fixed in a backwards-compatible fashion, and are not expected
to be in wide use.

Each warning should contain a link for more information on what the warning
means and how to resolve it.


To solve this problem, you can try the following approaches:


- If the issue is not solved by updating the dependencies, a fix has to be
implemented by those dependencies. You can help with that by notifying the
maintainers of this problem (e.g. by creating a bug report) or by proposing a
fix to the maintainers (e.g. by creating a pull request):

  - buf_redux@0.8.4
  - Repository: https://github.com/abonander/buf_redux
  - Detailed warning command: `cargo report future-incompatibilities --id 1 --package buf_redux@0.8.4`

  - multipart@0.18.0
  - Repository: http://github.com/abonander/multipart
  - Detailed warning command: `cargo report future-incompatibilities --id 1 --package multipart@0.18.0`

  - traitobject@0.1.0
  - Repository: https://github.com/reem/rust-traitobject.git
  - Detailed warning command: `cargo report future-incompatibilities --id 1 --package traitobject@0.1.0`

- If waiting for an upstream fix is not an option, you can use the `[patch]`
section in `Cargo.toml` to use your own version of the dependency. For more
information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section
```

Reply:
Thank you for pointing on this!
After the investigation, I found that the dependency graph is:
```
rosrust (latest) -> rouille (latest) ->  multipart (latest, not supported) -> buf_redux (latest, not supported)
                        -> hyper(lates)   ->  traitobject(latest)
```
There are plenty of issues on that:
https://github.com/tomaka/rouille/issues/271
https://github.com/reem/rust-traitobject/issues/8

As for us, I have the plan to do some refactoring on rosrust and propose a PR on that in the future, because rosrust has some implementation issues affecting bridge's performance that I would like to fix.